### DISCUSSION TITLE
Automatic Fallback to Public Servers

### DISCUSSION THREAD

**Discussion by HyP3r-** [rustdesk/rustdesk#2366] [General]
I have my own RustDesk relay server and enjoy using it with my friends and customers. However, it often happens that new customers and friends do not have my RustDesk relay server configuration when they install RustDesk or just start it. In this case it would be good if RustDesk first tries to connect with my relay server and then tries the public relay server.

**Comment by ChokunPlayZ** [comment]
I know this is almost a year old, but the option to connect with public server would be nice, maybe a button to click on "ID not found" error, or hold alt when click connect.

or you could build your own version of rustdesk with your server hard-coded, it is an option

**Comment by BigosIgos** [comment]
Bump

would appreciate such a feature.

**Comment by tschettervictor** [comment]
I believe there is such a feature now. Just add "@public" at the end of your ID string. This might require a download of 1.2.4 nightly version.

**Reply by BigosIgos** [reply]
and then it will fallback to the public server?

**Reply by tschettervictor** [reply]
It should yes.

**Reply by BigosIgos** [reply]
so if i have 2 endpoints i want to connect to. For example

PC1: Has a relay server set
PC2: Has not - uses Public

Client: I want to connect to both without messing with relayserver settings. 

This scenario would then be possible?

**Reply by tschettervictor** [reply]
It should be. You should be able to specify PC2@public to connect to it.

**Reply by BigosIgos** [reply]
thank you! will try :)

**Reply by 857074609** [reply]
> I believe there is such a feature now. Just add "@public" at the end of your ID string. This might require a download of 1.2.4 nightly version.<img width="479" height="95" alt="20251208065558" src="https://github.com/user-attachments/assets/151629c5-1d11-458b-890e-e6020463fa21" />  ？？？
Why does it ask me to log in?

**Comment by bilogic** [comment]
Do you all know the hostname of the public servers?

**Comment by Poordeveloper** [comment]
> it often happens that new customers and friends do not have my RustDesk relay server configuration when they install RustDesk or just start it

https://rustdesk.com/docs/en/self-host/client-configuration/#1-custom-client-generator-pro-only

**Comment by sirlouie** [comment]
Oh my goodness! I had no idea about the @public modifier. I had just made a similar suggestion for fallback, but was told no without elaboration, so I thought I'd search the forum a little more thoroughly to get some clarity. I'm glad I found this explanation, as it fits my needs! Thank you very much. 

One thing I noticed is that when I am on a privately-configured machine, and I have a publicly-configured machine in the address book (with @public), its status dot shows up as red, even though it can be remoted into. Can this be updated so that when @public is used in the ID field, the dot can show up green if the publicly-configured remote machine is available?

**Comment by shtefko** [comment]
I'm using self-hosted rustdesk server and the devices that are registered on that server (i.e. in their network settings they have my self-hosted server as ID and relay server) can only be reached through the self-hosted server. If the self-hosted server was down, I would not be able to reach any of these devices by appending @public to the ID. This only allows reaching clients that are registered on public servers.
It would be nice to have this fallback to public servers when self-hosted server was down, or to support some other method for high availability.

**Reply by scottharman** [reply]
Danger with this approach is that you might be running in a somewhat insecure manner. Ideally you'd pair it with a second factor (i.e. trust if using a self-hosted/private server, but recommend second factor (public key, totp) ... equally likely I'm overthinking it.
